.github/workflows/test-build-docker.yml

```diff
@@ -140,74 +87,209 @@ jobs:
+      with:
+        repository: opendistro-for-elasticsearch/index-management
+        ref: ${{env.p_tag_ism}}
+
+    - name: Checkout Alerting
+      uses: actions/checkout@v1
+      with:
+        repository: opendistro-for-elasticsearch/alerting
+        ref: ${{env.p_tag_alerting}}
+
+    - name: Checkout SQL
+      uses: actions/checkout@v1
+      with:
+        repository: opendistro-for-elasticsearch/sql
+        ref: ${{env.p_tag_sql}}
+        
+    - name: Checkout k-NN
+      uses: actions/checkout@v1
+      with:
+        repository: opendistro-for-elasticsearch/k-NN
+        ref: ${{env.p_tag_knn}}
+  
+    - name: Checkout AD
+      uses: actions/checkout@v1
+      with:
+        repository: opendistro-for-elasticsearch/anomaly-detection
+        ref: ${{env.p_tag_ad}}
+       
+    - name: Configure AWS
+      uses: aws-actions/configure-aws-credentials@v1
+      with:
+        aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+        aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+        aws-region: us-east-1
+
+    - name: Set Up JDK ${{ matrix.java }}
+      uses: actions/setup-java@v1
+      with:
+        java-version: ${{ matrix.java }}
+
+    - name: Start ES Container without Security
       run: |
         ODFE_VER=`./bin/version-info --od`
-        DOCKER_NAME=odfe-test-$ODFE_VER
-        docker exec -t $DOCKER_NAME /bin/bash -c "echo ''  >> /usr/share/elasticsearch/config/elasticsearch.yml"
-        docker exec -t $DOCKER_NAME /bin/bash -c "echo path.repo: [\"/usr/share/elasticsearch\"] >> /usr/share/elasticsearch/config/elasticsearch.yml"
-        docker restart $DOCKER_NAME
+        echo "ODFE VERSION $ODFE_VER"
         cd ..
+        DOCKER_NAME=odfe-test-$ODFE_VER
+        echo "Running newly tagged image with no security"
+        echo "FROM opendistroforelasticsearch/opendistroforelasticsearch:$ODFE_VER" >> Dockerfile
+        echo "RUN /usr/share/elasticsearch/bin/elasticsearch-plugin remove opendistro_security" >> Dockerfile
+        echo "RUN echo ''  >> /usr/share/elasticsearch/config/elasticsearch.yml" >> Dockerfile
+        echo "RUN echo \"path.repo: [\\\"/usr/share/elasticsearch\\\"]\" >> /usr/share/elasticsearch/config/elasticsearch.yml" >> Dockerfile
+        docker build -t odfe-http:no-security .
+        sleep 5
+ 
+        docker run -p 9200:9200 -d -p 9600:9600 -e "discovery.type=single-node" --name $DOCKER_NAME odfe-http:no-security
         sleep 30
-        curl -XGET http://localhost:9200/_cat/plugins > plugins.out
-        
-        if cat plugins.out|grep opendistro_index_management
-        then
-          cd index-management
-          ./gradlew integTest -Dtests.rest.cluster=localhost:9200 -Dtests.cluster=localhost:9200 -Dtests.clustername=es-integrationtest
-        else
-          echo "Skipping Test as Plugin not available"
-        fi
+        curl -XGET http://localhost:9200/_cat/plugins 
+
+    - name: Test ISM Plugin
+      if: always()
```

Comment:
Is this necessary?

Reply:
Actually it's optional for the first plugin. It's required for others, otherwise, if any step fails, the following step will be skipped.